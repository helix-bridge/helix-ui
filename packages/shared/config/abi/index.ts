import { AbiItem } from 'web3-utils';
import bankABI from './bankABI.json';
import bankErc20ABI from './bankErc20.json';
import bankErc20HelperABI from './bankErc20Helper.json';
import ktonABI from './ktonABI.json';
import tokenABI from './tokenABI.json';
import registryABI from './registryABI.json';
import tokenIssuingABI from './tokenIssuingABI.json';
import Erc20Byte32ABI from './Erc20-bytes32.json';
import Erc20StringABI from './Erc20-string.json';
import Erc20ABI from './Erc20.json';
import Erc20MappingTokenABI from './Erc20MappingToken.json';
import S2SMappingTokenABI from './S2SMappingToken.json';
import relayABI from './relay.json';

type keys =
  | 'ktonABI'
  | 'tokenABI'
  | 'registryABI'
  | 'bankABI'
  | 'tokenIssuingABI'
  | 'bankErc20ABI'
  | 'Erc20ABI'
  | 'Erc20StringABI'
  | 'Erc20Byte32ABI'
  | 'Erc20MappingTokenABI'
  | 'S2SMappingTokenABI'
  | 'relayABI';

export const abi = {
  bankABI,
  bankErc20ABI,
  bankErc20HelperABI,
  Erc20ABI,
  Erc20Byte32ABI,
  Erc20StringABI,
  ktonABI,
  Erc20MappingTokenABI,
  registryABI,
  relayABI,
  S2SMappingTokenABI,
  tokenABI,
  tokenIssuingABI,
} as {
  [key in keys]: AbiItem[];
};
