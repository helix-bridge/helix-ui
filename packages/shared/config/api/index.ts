export * from './darwinia';
export * from './evolution';
export * from './subscan';
