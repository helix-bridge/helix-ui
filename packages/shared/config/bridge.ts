import { ethereumDarwinia, ropstenPangolin } from './bridges/ethereum-darwinia';
import { ethereumCrabDVM, ropstenPangolinDVM } from './bridges/ethereum-dvm';
import { crabCrabDVM, pangolinPangolinDVM } from './bridges/substrate-dvm';
import { darwiniaCrabDVM, pangoroPangolinDVM } from './bridges/substrate-substrateDVM';

export const BRIDGES = [
  crabCrabDVM,
  darwiniaCrabDVM,
  ethereumCrabDVM,
  ethereumDarwinia,
  pangolinPangolinDVM,
  pangoroPangolinDVM,
  ropstenPangolin,
  ropstenPangolinDVM,
];

export { darwiniaCrabDVM, pangoroPangolinDVM };
