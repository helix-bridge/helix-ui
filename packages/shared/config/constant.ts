/* eslint-disable no-magic-numbers */
export const LONG_DURATION = 10 * 1000;

export const MIDDLE_DURATION = 5 * 1000;

export const SHORT_DURATION = 3 * 1000;

export const DATE_FORMAT = 'yyyy/MM/dd';

// export const DATE_TIME_FORMAT = 'yyyy/MM/dd HH:mm:ss';

export enum Path {
  root = '/',
  register = '/register',
  transaction = '/transaction',
}

export enum RegisterStatus {
  unregister,
  registered,
  registering,
}

export enum CrossChainStatus {
  pending,
  success,
  reverted,
}

export enum CrossChainStatusColor {
  '#00b3ff',
  '#00AA76',
  '#EC9D00',
}
