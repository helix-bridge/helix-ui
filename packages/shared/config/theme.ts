import { NetworkThemeConfig } from '../model';
import dark from '../theme/antd/dark.json';
import light from '../theme/antd/light.json';
import vars from '../theme/antd/vars.json';
import crab from '../theme/network/crab.json';
import crabDark from '../theme/network/dark/crab.json';
import darwiniaDark from '../theme/network/dark/darwinia.json';
import pangolinDark from '../theme/network/dark/pangolin.json';
import pangoroDark from '../theme/network/dark/pangoro.json';
import darwinia from '../theme/network/darwinia.json';
import pangolin from '../theme/network/pangolin.json';
import pangoro from '../theme/network/pangoro.json';

export const NETWORK_LIGHT_THEME: NetworkThemeConfig<{ [key in keyof typeof darwinia]: string }> = {
  crab,
  darwinia,
  pangolin,
  pangoro,
  ethereum: pangolin,
  ropsten: pangolin,
  tron: pangolin,
};

export const SKIN_THEME = {
  dark,
  light,
  vars,
};

export const NETWORK_DARK_THEME: NetworkThemeConfig<{ [key in keyof typeof darwiniaDark]: string }> = {
  crab: crabDark,
  darwinia: darwiniaDark,
  pangolin: pangolinDark,
  pangoro: pangoroDark,
  ethereum: pangolinDark,
  ropsten: pangolinDark,
  tron: pangolinDark,
};

export enum THEME {
  LIGHT = 'light',
  DARK = 'dark',
}
