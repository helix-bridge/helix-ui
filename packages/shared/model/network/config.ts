import { AddEthereumChainParameter } from '../metamask';
import { EthereumTypeNetwork, Network, NetworkCategory, PolkadotTypeNetwork } from './network';

export interface Facade {
  logo: string;
  logoMinor: string;
  logoWithText: string;
  logoSmart?: string;
  logoAssist2?: string;
}

interface DVMConfig {
  ring: string;
  kton: string;
  smartKton: string;
  smartRing: string;
  smartWithdrawRing: string;
  smartWithdrawKton: string;
  [key: string]: string;
}

interface ProviderConfig {
  rpc: string;
  etherscan: string;
}

export interface ChainConfig {
  facade: Facade;
  isTest: boolean;
  name: Network;
  provider: ProviderConfig;
  type: NetworkCategory[];
}

export interface EthereumChainConfig extends ChainConfig {
  name: EthereumTypeNetwork;
  ethereumChain: AddEthereumChainParameter;
}

export interface PolkadotChainConfig extends ChainConfig {
  name: PolkadotTypeNetwork;
  ss58Prefix: number;
  specVersion: number;
  endpoints: {
    mmr: string;
  };
}

export interface DVMChainConfig extends Omit<EthereumChainConfig, 'name'>, PolkadotChainConfig {
  dvm: DVMConfig;
}
