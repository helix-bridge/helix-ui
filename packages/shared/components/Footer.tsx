import { CopyrightOutlined, GithubOutlined, MailOutlined, TwitterOutlined } from '@ant-design/icons';
import { Divider, Layout } from 'antd';
import { getYear } from 'date-fns';
import { LanguageProps } from 'shared/components/widget/Language';
import { ThemeSwitchProps } from 'shared/components/widget/ThemeSwitch';
import { THEME } from 'shared/config/theme';
import { useEffect, useState } from 'react';
import { useITranslation } from '../hooks/translation';

type FooterProps = LanguageProps & { className?: string } & ThemeSwitchProps;

export function Footer({ theme, className = '' }: FooterProps) {
  const { t } = useITranslation();
  const [mainnetOrTestnet, setMainnetOrTestnet] = useState<{ label: 'Mainnet' | 'Testnet'; link: string }>();

  useEffect(() => {
    if (window.location.hostname === 'helixbridge.app') {
      setMainnetOrTestnet({ label: 'Testnet', link: 'helix-apps-test.vercel.app' });
    } else if (window.location.hostname === 'helix-apps-test.vercel.app') {
      setMainnetOrTestnet({ label: 'Mainnet', link: 'helixbridge.app' });
    } else if (window.location.hostname === 'helix-stg.vercel.app') {
      setMainnetOrTestnet({ label: 'Testnet', link: 'helix-stg-test.vercel.app' });
    } else if (window.location.hostname === 'helix-stg-test.vercel.app') {
      setMainnetOrTestnet({ label: 'Mainnet', link: 'helix-stg.vercel.app' });
    }
  }, []);

  return (
    <Layout.Footer
      className={`flex items-center justify-between z-10 px-0 py-4 container ${className}`}
      style={{
        background: theme === THEME.LIGHT ? '#ccc' : 'transparent',
      }}
    >
      <div className="md:flex md:gap-4 md:flex-wrap text-gray-400">
        <span className="flex items-center justify-center">
          <CopyrightOutlined style={{ fontSize: '0.9rem' }} />
          <span className="ml-1 text-gray-400 text-base font-medium">
            {t('{{year}} Developed by Helix Team', { year: getYear(new Date()) })}
          </span>
        </span>
      </div>

      <div className="flex items-center gap-4">
        <a
          href="https://github.com/helix-bridge"
          target="_blank"
          rel="noreferrer"
          className="flex text-gray-400 hover:text-gray-200"
        >
          <GithubOutlined style={{ fontSize: '1.25rem' }} />
        </a>

        <a
          href="https://twitter.com/helixbridges"
          target="_blank"
          rel="noreferrer"
          className="flex text-gray-400 hover:text-gray-200"
        >
          <TwitterOutlined style={{ fontSize: '1.25rem' }} />
        </a>

        <a
          href="mailto:hello@helixbridge.app"
          target="_blank"
          rel="noreferrer"
          className="flex text-gray-400 hover:text-gray-200"
        >
          <MailOutlined style={{ fontSize: '1.25rem' }} />
        </a>

        {mainnetOrTestnet && (
          <>
            <div className="w-[1px] h-3 bg-white/30" />
            <a
              className="text-gray-400 text-sm font-bold hover:underline"
              target="_blank"
              rel="noreferrer"
              href={mainnetOrTestnet.link}
            >
              {mainnetOrTestnet.label}
            </a>
          </>
        )}

        <Divider
          type="vertical"
          style={{
            margin: 0,
            borderColor: 'rgba(255, 255, 255, .3)',
            height: '0.5em',
            marginTop: '0.25em',
            display: 'none',
          }}
        />
      </div>
    </Layout.Footer>
  );
}
