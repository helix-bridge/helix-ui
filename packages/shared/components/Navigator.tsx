import { DownOutlined } from '@ant-design/icons';
import { AntdIconProps } from '@ant-design/icons/lib/components/AntdIcon';
import { Dropdown, Menu, Typography } from 'antd';
import Link from 'next/link';
import { useRouter } from 'next/router';
import React from 'react';
import { useMemo } from 'react';
import { isDev } from 'shared/config/env';
import { THEME } from 'shared/config/theme';
import { useITranslation } from '../hooks';

export interface Nav {
  label: string;
  path: string;
  extra?: boolean;
  hide?: boolean;
  Icon?: (props: AntdIconProps) => JSX.Element;
  className?: string;
}

interface NavigatorProps {
  navigators: Nav[];
  theme?: THEME;
  onClick?: () => void;
}

function NavLink({ nav, theme }: { nav: Nav; theme: THEME }) {
  const { t } = useITranslation();
  const router = useRouter();
  const textCls = useMemo(() => (theme === 'dark' ? 'text-white' : 'text-pangolin-main'), [theme]);
  const active =
    nav.path === router.pathname
      ? theme === 'dark'
        ? 'shadow-mock-bottom-border-light'
        : 'shadow-mock-bottom-border'
      : '';

  return (
    <div
      onClick={() => {
        if (nav.extra) {
          window.open(nav.path, '_blank');
        } else if (nav.path) {
          router.push(nav.path);
        } else {
          // nothing
        }
      }}
      className={`${active} transition-all duration-300 ease-in-out opacity-100 hover:opacity-80 cursor-pointer whitespace-nowrap`}
      key={nav.label}
    >
      <span className={`text-base font-normal ${textCls}`}>{t(nav.label)}</span>
    </div>
  );
}

function RouteLink({ path, label }: Nav) {
  const { t } = useITranslation();

  return path.includes('http') ? (
    <Typography.Link href={path} target="_blank">
      {t(label)}
    </Typography.Link>
  ) : (
    <Link href={path}>{t(label)}</Link>
  );
}

export const getActiveNav = (path: string, navigators: Nav[]) => {
  return navigators
    .filter((item) => path === item.path)
    .map((item) => navigators.find((nav) => nav.path.startsWith(item.path)))
    .filter((item) => !!item) as Nav[];
};

export function Navigator({ navigators, onClick, theme = THEME.DARK }: NavigatorProps) {
  const { t } = useITranslation();
  const { pathname } = useRouter();

  const navItems = useMemo(() => navigators.filter((item) => isDev || (!isDev && !item.hide)), [navigators]);

  const selectedNavMenu = useMemo<string[]>(() => {
    const nav = getActiveNav(pathname, navigators);

    return [nav.length ? nav[0].path : ''];
  }, [pathname, navigators]);

  return (
    <>
      <div className="hidden lg:flex items-center gap-8 ml-8">
        {navItems.map((nav, index) =>
          Array.isArray(nav) ? (
            <Dropdown
              key={index}
              overlay={
                <Menu
                  items={nav.slice(1).map((item, idx) => ({
                    key: item.path + '_' + idx,
                    label: <RouteLink {...item} />,
                  }))}
                />
              }
            >
              <div className="flex items-center">
                <NavLink nav={nav[0]} theme={theme} />
                <DownOutlined className="ml-2" />
              </div>
            </Dropdown>
          ) : (
            <NavLink nav={nav} key={index} theme={theme} />
          )
        )}
      </div>

      <Menu
        theme={theme}
        mode="inline"
        defaultSelectedKeys={selectedNavMenu}
        items={navItems.map(({ Icon, path, label, className }) => ({
          key: path,
          label: <Link href={path}>{t(label)}</Link>,
          icon: Icon ? <Icon /> : null,
          className,
          onClick() {
            if (onClick) {
              onClick();
            }
          },
        }))}
        className="block lg:hidden flex-1"
        style={{ background: theme === THEME.DARK ? 'transparent' : 'inherit' }}
      />
    </>
  );
}
