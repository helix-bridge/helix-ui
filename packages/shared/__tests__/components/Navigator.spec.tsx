/// <reference types="jest" />

import { create } from 'react-test-renderer';
import { Navigator } from '../../components/Navigator';
import { THEME } from '../../config/theme';

jest.mock('next-i18next', () => ({
  useTranslation() {
    return {
      t: (label: string) => label,
    };
  },
}));

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

describe('<Navigator />', () => {
  it('render navigator', () => {
    const component = create(
      <Navigator
        navigators={[
          { label: 'Home', path: '/' },
          { label: 'History', path: '/history' },
        ]}
        theme={THEME.DARK}
      />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
