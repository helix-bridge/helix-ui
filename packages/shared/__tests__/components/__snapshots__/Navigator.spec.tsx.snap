// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigator /> render navigator 1`] = `
Array [
  <div
    className="hidden lg:flex items-center gap-8 ml-8"
  >
    <div
      className=" transition-all duration-300 ease-in-out opacity-100 hover:opacity-80 cursor-pointer whitespace-nowrap"
      onClick={[Function]}
    >
      <span
        className="text-base font-normal text-white"
      >
        Home
      </span>
    </div>
    <div
      className=" transition-all duration-300 ease-in-out opacity-100 hover:opacity-80 cursor-pointer whitespace-nowrap"
      onClick={[Function]}
    >
      <span
        className="text-base font-normal text-white"
      >
        History
      </span>
    </div>
  </div>,
  <ul
    className="ant-menu ant-menu-root ant-menu-inline ant-menu-dark block lg:hidden flex-1"
    data-menu-list={true}
    onKeyDown={[Function]}
    role="menu"
    style={
      Object {
        "background": "transparent",
      }
    }
    tabIndex={0}
  >
    <li
      className="ant-menu-item ant-menu-item-only-child"
      data-menu-id={null}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </span>
    </li>
    <li
      className="ant-menu-item ant-menu-item-only-child"
      data-menu-id={null}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        <a
          href="/history"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          History
        </a>
      </span>
    </li>
  </ul>,
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;
